### ISSUE TITLE
Can't get the docker to run

### ISSUE THREAD

**Issue by TomerFi** [tomplus/qnap-git-server#1]
Hello
Awesome project, seems very useful.
Thank you.

I can't seem to get it to work though, No matter what I do, I keep getting the following error in the console:
``` text
standard_init_linux.go:187: exec user process caused "exec format error"   
```

Do you have any ideas for what can I do to get it to work?

**Comment by tomplus** [comment]
Hi @TomerFi Thanks, I'm glad that you want to use it :)

This error appears if you try to start an image which was prepared for another architecture and their "exec format" is incorrect (32bit vs 64bit, arm vs intel etc.). This image https://hub.docker.com/r/tpimages/qnap-git-server/ is prepared for arm32v7 (I use TS-228). Which platform do you have ? 

You can try to clone/copy this repo to your qnap, fix base image (first line of Dockerfile `FROM arm32v7/ubuntu:14.04`) and [build](https://github.com/tomplus/qnap-git-server#building) it. I hope it will help.

**Comment by TomerFi** [comment]
Yes that's probably my problem.
I'm using TS251A which is 64bit.

**Comment by tomplus** [comment]
Try to replace from `arm32v7` to `amd64` in the Dockerfile and build your image.

**Comment by TomerFi** [comment]
Thank you very much.
I will give that a try and post back the result.

**Comment by TomerFi** [comment]
I'm sorry, I'm still a beginner in linux and dockers.
I'm not sure what I'm doing wrong here.
After changing `arm32v7` to `amd64`, the Container Station failed to create the container, in the console all I see is `Error grabbing logs: EOF `.

**Comment by TomerFi** [comment]
Got it running!! :-)

In ContainerStation, when creating the container from the local image I:
- Configured the entry point as /bin/sh - Which got rid of the `EOF` message and showed me the real problem, I got the message `User git does not have valid homedir !` from your script, and the reason for it was:
- I was unable to map the host's `share/git/` folder to the container's `home/git` because the gui doesn't allow typing the source, only selecting it from the shared folders is allowed. So I did all the folder preparation in `share/Public/git` instead of `share/git` and was able to map it to the container's `home/git/`.

So... it's running now. :-)
Now, I'll play with it a bit and confirm that everything is working well.

Thank you very much!
